js/indicators/tema.src.js

```diff
@@ -0,0 +1,235 @@
+     * @since 7.0.0
+     * @optionparent plotOptions.tema
+     */
+    {}, {
+        getValues: function (series, params) {
+            var period = params.period,
+                doubledPeriod = 2 * period,
+                tripledPeriod = 3 * period,
+                xVal = series.xData,
+                yVal = series.yData,
+                yValLen = yVal ? yVal.length : 0,
+                EMApercent = (2 / (period + 1)),
+                index = -1,
+                sum = 0,
+                ema = 0,
+                SMA = 0,
+                EmaArr = [],
+                emaEMAArr = [],
+                TEMA = [],
+                xDataTema = [],
+                yDataTema = [],
+                calEMA,
+                calDEMA,
+                calTEMA,
+                emaEMA,
+                emaEMAEMA,
+                i,
+                TEMAPoint;
+
+            // Check period, if bigger than EMA points length, skip
+            if (yValLen < 3 * period - 2) {
+                return false;
+            }
+
+            // Switch index for OHLC / Candlestick / Arearange
+            if (isArray(yVal[0])) {
+                index = params.index ? params.index : 0;
+            }
+
+            // Accumulate first N-points
+            sum = EMA.prototype.accumulatePeriodPoints(
+              period,
+              index,
+              yVal
+            );
+
+            // first point
+            SMA = sum / period;
+            sum = 0;
+
+            // Calculate value one-by-one for each period in visible data
+            for (i = period; i < yValLen + 1; i++) {
+                ema = EMA.prototype.calculateEma(
+                    xVal,
+                    yVal,
+                    i,
+                    EMApercent,
+                    calEMA,
+                    index,
+                    SMA
+                )[1];
+                EmaArr.push(ema);
+                calEMA = ema;
+
+                // Summing first period points for ema(ema)
+                if (i < doubledPeriod) {
+                    sum += ema;
+                }
+
+                // Calculate dema
+                if (i >= (doubledPeriod)) {
+                    // First dema point
+                    if (i === (doubledPeriod)) {
+                        SMA = sum / period;
+                        sum = 0;
+                    }
+                    ema = EmaArr[i - period - 1];
+                    emaEMA = EMA.prototype.calculateEma(
+                      [],
+                      [ema],
+                      1,
+                      EMApercent,
+                      calDEMA,
+                      -1,
+                      SMA
+                    )[1];
+                    emaEMAArr.push(emaEMA);
+                    calDEMA = emaEMA;
+                    // Summing first period points for ema(ema(ema))
+                    if (i < tripledPeriod) {
+                        sum += emaEMA;
+                    }
```

Comment:
Why you check it here, instead of out of doublePeriod condition? Should it not be like line 95-97 ?

Reply:
I feel like all `if`s from 94-148 could be optimized - let's discuss this when you have time @raf18seb @sebastianbochan